src/unix/linux.c

```diff
@@ -1179,12 +1179,22 @@ static void uv__get_cgroup1_memory_limits(char buf[static 1024], uint64_t* high,
      * as indicated by uv__read_uint64 returning 0.
      */
      if (*high != 0 && *max != 0)
-       return;
+       goto update_limits;
   }
 
   /* Fall back to the limits of the global memory controller. */
   *high = uv__read_uint64("/sys/fs/cgroup/memory/memory.soft_limit_in_bytes");
   *max = uv__read_uint64("/sys/fs/cgroup/memory/memory.limit_in_bytes");
+
+  /* uv__read_uint64 detects cgroup2's "max", so we need to separately detect
+   * cgroup1's maximum value (which is derived from LONG_MAX and PAGE_SIZE).
+   */
+update_limits:
+  uint64_t cgroup1_max = LONG_MAX & ~(sysconf(_SC_PAGESIZE) - 1);
```

Comment:
The c standard forbids declaring variables after a `:`. The C89 style guide requires them to be at the top of the function
```
/__w/libuv/libuv/src/unix/linux.c:1193:3: error: expected expression
[37](https://github.com/libuv/libuv/actions/runs/3949084551/jobs/6759859025#step:6:38)
  uint64_t cgroup1_max = LONG_MAX & ~(sysconf(_SC_PAGESIZE) - 1);
[38](https://github.com/libuv/libuv/actions/runs/3949084551/jobs/6759859025#step:6:39)
  ^
[39](https://github.com/libuv/libuv/actions/runs/3949084551/jobs/6759859025#step:6:40)
```

Reply:
Right, of course; thanks. My local build must have been configured differently.